examples/amp-web-push/publisher-files/permission-dialog.html

```diff
@@ -0,0 +1,64 @@
+console.log.apply(null,n)}},e.prototype.onChannelMessageReceived=function(e){var n=e.data
+e.origin,e.source
+if(this.messages[n.id]&&n.isReply){var i=this.messages[n.id]
+delete this.messages[n.id]
+var o=i.promiseResolver
+i.message=n.data,this.debug&&this.log("Received reply for topic '"+n.topic+"':",n.data),o([n.data,this.sendReply.bind(this,n.id,i.topic)])}else{var t=this.listeners[n.topic]
+if(!t)return
+this.debug&&this.log("Received new message for topic '"+n.topic+"':",n.data)
+for(var s=0,r=t;s<r.length;s++){var c=r[s]
+c(n.data,this.sendReply.bind(this,n.id,n.topic))}}},e.prototype.on=function(e,n){this.listeners[e]?this.listeners[e].push(n):this.listeners[e]=[n]},e.prototype.off=function(e,n){if(n){var i=this.listeners.topic.indexOf(n);-1!==i&&this.listeners.splice(i,1)}else this.listeners[e]&&delete this.listeners[e]},e.prototype.sendReply=function(e,n,i){var o=this,t={id:e,topic:n,data:i,isReply:!0}
+return this.messagePort.postMessage(t),new Promise(function(e){o.messages[t.id]={message:i,topic:n,promiseResolver:e}})},e.prototype.send=function(e,n){var i=this,o={id:crypto.getRandomValues(new Uint8Array(10)).join(""),topic:e,data:n}
+return this.log("Sending "+e+":",n),this.messagePort.postMessage(o),new Promise(function(t){i.messages[o.id]={message:n,topic:e,promiseResolver:t}})},e}()
+"undefined"!=typeof module&&(module.exports=WindowMessenger)
+var AmpWebPushPermissionDialog=function(){function e(e){e&&e.debug&&(this.debug=!0),this.ampMessenger=new WindowMessenger({debug:this.debug})}return e.prototype.isCurrentDialogPopup=function(){return!!window.opener&&window.opener!==window},e.prototype.requestNotificationPermission=function(){return new Promise(function(e,n){try{Notification.requestPermission(function(n){return e(n)})}catch(i){n(i)}})},e.prototype.tryDecodeUriComponent=function(e,n){void 0===n&&(n="")
+try{return decodeURIComponent(e)}catch(i){return n}},e.prototype.parseQueryString=function(e){var n=Object.create(null)
+if(!e)return n
+for(var i,o=/(?:^[#?]?|&)([^=&]+)(?:=([^&]*))?/g;i=o.exec(e);){var t=this.tryDecodeUriComponent(i[1]).trim(),s=i[2]?this.tryDecodeUriComponent(i[2]).trim():""
+n[t]=s}return n},e.prototype.run=function(){var e=this
+if(this.isCurrentDialogPopup())this.ampMessenger.connect(opener,"*"),this.requestNotificationPermission().then(function(n){return e.ampMessenger.send(WindowMessenger.Topics.NOTIFICATION_PERMISSION_STATE,n)}).then(function(e){var n=e[0]
+e[1]
+n&&n.closeFrame&&window.close()})
+else{var n=this.parseQueryString(window.location.search)
+if(!n["return"])throw Error("Expecting return URL query parameter to redirect back.")
+this.requestNotificationPermission().then(function(i){window.location.href=e.tryDecodeUriComponent(n["return"])})}},e}()
+new AmpWebPushPermissionDialog({debug:!0}).run()
+  </script>
+</head>
+<body>
+  <p class="centered" id="subscribe-message">Click Allow to subscribe.</p>
```

Comment:
We actually think it's best to ship the extension without this text at all, and instead provide users a working example of what they could add. That way, we're not encouraging any specific kind of implementation. For the sample CSS we have in amp-web-push.css, we'd like to take that out as well and provide it as an example implementation, but not ship the base plugin with this default style.